File: 原創不分類題庫/Contents/d632/d632.py
import sys

for a in sys.stdin:
    a = a.strip()

    b= sys.stdin.readline().strip()

    carry = 0
    i = 31
    result = []
    while i >= 0:
        temp = int(a[i]) + int(b[i]) + carry
        result.append(temp % 2)
        carry = temp // 2
        i -= 1
    
    print(a)
    print(b)
    print("---------------------------------")
    result = result[::-1]
    for i in range(32):
        print(result[i], end="", sep="")
    print("\n****End of Data******************")

File: 原創不分類題庫/Contents/d632/d632.c
#include<stdio.h>

int main(void)
{
	char a[33], b[33];
	int result[33];
	int temp;
	
	while(scanf("%s", a) != EOF)
	{
		scanf("%s", b);
		
		int carry = 0;
		for(int i = 31; i >= 0; i--)
		{
			temp = (a[i] - '0') + (b[i] - '0') + carry;
			result[i] = temp % 2;
			carry = temp / 2;
		}
		
		puts(a);
		puts(b);
		printf("---------------------------------\n");
		for(int i = 0; i < 32; i++)
			printf("%d", result[i]);
		printf("\n****End of Data******************\n");
	}
	
	return 0;
}
